alacritty_terminal/src/config/bindings.rs

```diff
@@ -71,113 +71,158 @@ pub fn default_key_bindings() -> Vec<KeyBinding> {
-        Key::F9, [alt: true]; Action::Esc("\x1b[20;6~".into());
-        Key::F10, [alt: true]; Action::Esc("\x1b[21;6~".into());
-        Key::F11, [alt: true]; Action::Esc("\x1b[23;6~".into());
-        Key::F12, [alt: true]; Action::Esc("\x1b[24;6~".into());
-        Key::F1, [logo: true]; Action::Esc("\x1b[1;3P".into());
-        Key::F2, [logo: true]; Action::Esc("\x1b[1;3Q".into());
-        Key::F3, [logo: true]; Action::Esc("\x1b[1;3R".into());
-        Key::F4, [logo: true]; Action::Esc("\x1b[1;3S".into());
-        Key::F5, [logo: true]; Action::Esc("\x1b[15;3~".into());
-        Key::F6, [logo: true]; Action::Esc("\x1b[17;3~".into());
-        Key::F7, [logo: true]; Action::Esc("\x1b[18;3~".into());
-        Key::F8, [logo: true]; Action::Esc("\x1b[19;3~".into());
-        Key::F9, [logo: true]; Action::Esc("\x1b[20;3~".into());
-        Key::F10, [logo: true]; Action::Esc("\x1b[21;3~".into());
-        Key::F11, [logo: true]; Action::Esc("\x1b[23;3~".into());
-        Key::F12, [logo: true]; Action::Esc("\x1b[24;3~".into());
+        Key::F13; Action::Esc("\x1b[25~".into());
+        Key::F14; Action::Esc("\x1b[26~".into());
+        Key::F15; Action::Esc("\x1b[28~".into());
+        Key::F16; Action::Esc("\x1b[29~".into());
+        Key::F17; Action::Esc("\x1b[31~".into());
+        Key::F18; Action::Esc("\x1b[32~".into());
+        Key::F19; Action::Esc("\x1b[33~".into());
+        Key::F20; Action::Esc("\x1b[34~".into());
         Key::NumpadEnter; Action::Esc("\n".into());
     );
 
+    //   Code     Modifiers
+    // ---------+---------------------------
+    //    2     | Shift
+    //    3     | Alt
+    //    4     | Shift + Alt
+    //    5     | Control
+    //    6     | Shift + Control
+    //    7     | Alt + Control
+    //    8     | Shift + Alt + Control
+    //    9     | Meta
+    //    10    | Meta + Shift
+    //    11    | Meta + Alt
+    //    12    | Meta + Alt + Shift
+    //    13    | Meta + Ctrl
+    //    14    | Meta + Ctrl + Shift
+    //    15    | Meta + Ctrl + Alt
+    //    16    | Meta + Ctrl + Alt + Shift
+    // ---------+---------------------------
+    //
+    // from: https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-PC-Style-Function-Keys
+    let modifiers = vec![
+        ModifiersState { shift: true, ..ModifiersState::default() },
+        ModifiersState { alt: true, ..ModifiersState::default() },
+        ModifiersState { shift: true, alt: true, ..ModifiersState::default() },
+        ModifiersState { ctrl: true, ..ModifiersState::default() },
+        ModifiersState { shift: true, ctrl: true, ..ModifiersState::default() },
+        ModifiersState { alt: true, ctrl: true, ..ModifiersState::default() },
+        ModifiersState { shift: true, alt: true, ctrl: true, ..ModifiersState::default() },
+    ];
+
+    for (modifiers_code, mods) in modifiers.iter().enumerate() {
+        let modifiers_code = modifiers_code + 2;
```

Comment:
This might be clearer if you don't reuse `modifiers_code`, since technically, until you shadow the binding introduced from the `for` it's inaccurate. Instead I might write:

```rust
let modifiers_code = index + 2;
```

Very small point tho.